lib/internal/modules/esm/translators.js

```diff
@@ -350,12 +350,17 @@ translators.set('commonjs', async function commonjsStrategy(url, source,
     }
   } : loadCJSModule;
 
-  try {
-    // We still need to read the FS to detect the exports.
-    source ??= readFileSync(new URL(url), 'utf8');
-  } catch {
-    // Continue regardless of error.
+  const urlPath = URL.parse(url);
+
+  if (urlPath != null) {
+    try {
+      // We still need to read the FS to detect the exports.
+      source ??= readFileSync(urlPath, 'utf8');
+    } catch {
+      // Continue regardless of error.
+    }
```

Comment:
Same here, also I don't think `URL.parse(url)` can ever be null at this point

Reply:
Any downside to have this check either way?